conf/init/ol-solr-updater.conf

```diff
@@ -6,5 +6,6 @@ stop on runlevel [!2345]
 
 chdir /openlibrary
 respawn
```

Comment:
You want both. The first says to respawn, the second sets a limit. See: http://upstart.ubuntu.com/cookbook/#respawn and http://upstart.ubuntu.com/cookbook/#respawn-limit